File: database/database-annotation-processor/src/main/java/ru/tinkoff/kora/database/annotation/processor/jdbc/JdbcTypes.java
package ru.tinkoff.kora.database.annotation.processor.jdbc;

import com.squareup.javapoet.ClassName;

public final class JdbcTypes {

    private JdbcTypes() { }

    public static final ClassName CONNECTION = ClassName.get("java.sql", "Connection");
    public static final ClassName CONNECTION_FACTORY = ClassName.get("ru.tinkoff.kora.database.jdbc", "JdbcConnectionFactory");
    public static final ClassName JDBC_REPOSITORY = ClassName.get("ru.tinkoff.kora.database.jdbc", "JdbcRepository");

    public static final String RESULT_PACKAGE = "ru.tinkoff.kora.database.jdbc.mapper.result";
    public static final ClassName RESULT_SET_MAPPER = ClassName.get(RESULT_PACKAGE, "JdbcResultSetMapper");
    public static final ClassName ROW_MAPPER = ClassName.get(RESULT_PACKAGE, "JdbcRowMapper");
    public static final ClassName RESULT_COLUMN_MAPPER = ClassName.get(RESULT_PACKAGE, "JdbcResultColumnMapper");

    public static final String PARAMETER_PACKAGE = "ru.tinkoff.kora.database.jdbc.mapper.parameter";
    public static final ClassName PARAMETER_COLUMN_MAPPER = ClassName.get(PARAMETER_PACKAGE, "JdbcParameterColumnMapper");

}

File: database/database-symbol-processor/src/main/kotlin/ru/tinkoff/kora/database/symbol/processor/jdbc/JdbcTypes.kt
package ru.tinkoff.kora.database.symbol.processor.jdbc

import com.squareup.kotlinpoet.ClassName

object JdbcTypes {
    val connection = ClassName("java.sql", "Connection")
    val resultSet = ClassName("java.sql", "ResultSet")
    val connectionFactory = ClassName("ru.tinkoff.kora.database.jdbc", "JdbcConnectionFactory")
    val jdbcRepository = ClassName("ru.tinkoff.kora.database.jdbc", "JdbcRepository")
    val jdbcResultSetMapper = ClassName("ru.tinkoff.kora.database.jdbc.mapper.result", "JdbcResultSetMapper")
    val jdbcRowMapper = ClassName("ru.tinkoff.kora.database.jdbc.mapper.result", "JdbcRowMapper")
    val jdbcResultColumnMapper = ClassName("ru.tinkoff.kora.database.jdbc.mapper.result", "JdbcResultColumnMapper")

    val jdbcParameterColumnMapper = ClassName("ru.tinkoff.kora.database.jdbc.mapper.parameter", "JdbcParameterColumnMapper")
}
